src/gaxios.ts

```diff
@@ -219,21 +232,25 @@ export class Gaxios {
     if (opts.data) {
       if (isStream.readable(opts.data)) {
         opts.body = opts.data;
+      } else if (hasBuffer() && Buffer.isBuffer(opts.data)) {
+        // Do not attempt to JSON.stringify() a Buffer:
+        opts.body = opts.data;
+        if (!hasHeader(opts, 'Content-Type')) {
+          opts.headers['Content-Type'] = 'application/json';
+        }
       } else if (typeof opts.data === 'object') {
-        opts.body = JSON.stringify(opts.data);
-        // Allow the user to specifiy their own content type,
-        // such as application/json-patch+json; for historical reasons this
-        // content type must currently be a json type, as we are relying on
-        // application/x-www-form-urlencoded (which is incompatible with
-        // upstream GCP APIs) being rewritten to application/json.
-        //
-        // TODO: refactor upstream dependencies to stop relying on this
-        // side-effect.
+        // If www-form-urlencoded content type has been set, but data is
+        // provided as an object, serialize the content using querystring:
         if (
-          !opts.headers['Content-Type'] ||
-          !opts.headers['Content-Type'].includes('json')
+          opts.headers['Content-Type'] &&
```

Comment:
Why do you have to do the non-null check here, if you're just comparing it to a specific string below?  Thinking you can delete line 245.